### PR TITLE
test(calculate): verify streak formulas for different starting days

### LINKED ISSUE #1476: test(calculate): verify streak formulas for different starting days of the week timeline (Variation 1)
## 🧩 Background
Streak computation calculations are susceptible to off-by-one errors when managing calendar offsets and date boundaries.

## 🎯 Objective
Add a specific test simulating weeks starting on Sunday versus Monday formats.

## 📁 Files to touch
- `lib/calculate.test.ts`

## 🛠️ Implementation steps
- Configure custom contribution data matching this pattern.
- Execute the streak calculator and assert accurate currents and longest values.

## ✅ Definition of done
- [ ] 1 new test case simulating this specific timeline.
- [ ] All streak mathematical assertions are accurate.

### PR DESCRIPTION
**Pull request by vivekktrivedi06-hue**
## Description
Adds test cases to verify streak calculations for different starting days of the week.

Fixes #1476

## Pillar
- [ ] 🎨 Pillar 1 – New Theme Design
- [ ] 🔷 Pillar 2 – Geometric SVG Improvement
- [x] 🕒 Pillar 3 – Timezone Logic Optimization
- [ ] 🛠 Other (Bug fix, refactoring, docs)

## Type of change
- [x] test

## Checklist
- [x] I have run npm run format
- [x] I have run npm run lint
- [x] I have run npm run test
- [x] Tests are passing locally

### CODE REVIEW

**Comment by vivekktrivedi06-hue** [comment]
Hi @maintainer 👋

All required checks (lint, format, tests, and build) are passing successfully.

The only failing check is related to Vercel deployment authorization, which requires maintainer approval.

Kindly review the PR when possible. Thanks!

File: lib/calculate.test.ts [hunk 1]
```diff
@@ -527,200 +527,28 @@ describe('calculateStreak — empty and sparse year edge cases', () => {
...
+    1,
+    1,
+    1,
+    1,
+    1,
+    1,
+    1, // week 2
+  ]);
 
-describe('calculateStreak — todayDate format', () => {
```
Comment (JhaSourav07) [inline lib/calculate.test.ts:1403]: why removed this ?

**Review by Aamod007 — CHANGES_REQUESTED** [review]
deletes 194 lines of existing tests — 4 complete describe blocks:

calculateStreak — todayDate format (3 tests)
calculateStreak — year boundary transition (1 test)
aggregateCalendars (3 tests)
calculateWrappedStats (6 tests)

**Comment by vivekktrivedi06-hue** [comment]
Hi @JhaSourav07 👋

Thanks for pointing this out.

This change was not intentional — it got overwritten while resolving merge conflicts and rebasing with the latest main branch.

I’ve now ensured the required logic/tests are preserved.
Also, instead of the previous generic format test, I’ve added more specific assertions that validate todayDate across different timezone scenarios (including exact expected values), which ensures stronger coverage.

Let me know if you'd still prefer keeping the original format-specific test as well.
Please let me know if any specific part needs to be restored or improved further. Thanks!

**Comment by vivekktrivedi06-hue** [comment]
Hi @maintainers 👋

The CI is failing due to an unrelated issue in `app/page.tsx`:

Error:
Cannot find name 'InteractiveViewer'

This file is not part of my changes. Kindly review or fix this upstream.

My changes are limited to `lib/calculate.test.ts`.

Thanks!

**Comment by vivekktrivedi06-hue** [comment]
Thanks for resolving the issue! 🙌

All CI checks are now passing. The remaining Vercel check seems to require authorization and is unrelated to my changes.

Ready for review/merge 🚀

**Review by Aamod007 — APPROVED** [review]
